### ISSUE TITLE
TestCluster_Open failed on CircleCI

### ISSUE THREAD

**Issue by otoolep** [influxdb/influxdb#3144]
```
=== RUN TestCluster_Open
2015/06/25 21:20:33 [DEBUG] raft-net: 127.0.0.1:58795 accepted connection from: 127.0.0.1:44249
2015/06/25 21:20:33 [DEBUG] raft-net: 127.0.0.1:50409 accepted connection from: 127.0.0.1:44585
2015/06/25 21:20:33 [DEBUG] raft-net: 127.0.0.1:50409 accepted connection from: 127.0.0.1:44588
2015/06/25 21:20:33 [DEBUG] raft-net: 127.0.0.1:58795 accepted connection from: 127.0.0.1:44253
panic: failed to save current term: database not open

goroutine 131 [running]:
github.com/hashicorp/raft.(*Raft).setCurrentTerm(0xc2080ba1c0, 0x1)
    /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/raft.go:1576 +0x1fe
github.com/hashicorp/raft.(*Raft).appendEntries(0xc2080ba1c0, 0x9a9880, 0xc2081716d0, 0x0, 0x0, 0xc20818b080, 0xc2081716d0)
    /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/raft.go:1232 +0x1821
github.com/hashicorp/raft.(*Raft).processRPC(0xc2080ba1c0, 0x9a9880, 0xc2081716d0, 0x0, 0x0, 0xc20818b080)
    /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/raft.go:1175 +0x1bb
github.com/hashicorp/raft.(*Raft).runFollower(0xc2080ba1c0)
    /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/raft.go:589 +0x333
github.com/hashicorp/raft.(*Raft).run(0xc2080ba1c0)
    /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/raft.go:572 +0xc5
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.run)Â·fm()
    /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/raft.go:241 +0x35
github.com/hashicorp/raft.funcÂ·011()
    /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/state.go:152 +0x80
created by github.com/hashicorp/raft.(*raftState).goFunc
```

There was some question about whether stuff was being closed in the wrong order, but this does not appear to be the case.

**Comment by otoolep** [comment]
CC @benbjohnson

**Comment by otoolep** [comment]
Another failure:
```
--- FAIL: TestCluster_Open (1.19s)
panic: store: i=0, addr=127.0.0.1:34160, err=create local node: create node: exec failed: apply: leadership lost while committing log [recovered]
	panic: store: i=0, addr=127.0.0.1:34160, err=create local node: create node: exec failed: apply: leadership lost while committing log

goroutine 147 [running]:
testing.funcÂ·006()
	/usr/local/go/src/testing/testing.go:441 +0x249
github.com/influxdb/influxdb/meta_test.MustOpenCluster(0x3, 0x46bdf0)
	/home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/meta/store_test.go:1021 +0x568
github.com/influxdb/influxdb/meta_test.TestCluster_Open(0xc2080e1d40)
	/home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/meta/store_test.go:783 +0x35
testing.tRunner(0xc2080e1d40, 0xedf830)
	/usr/local/go/src/testing/testing.go:447 +0x134
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xd4e

goroutine 1 [chan receive]:
testing.RunTests(0xceaea8, 0xedee40, 0x6d, 0x6d, 0x1)
	/usr/local/go/src/testing/testing.go:556 +0xdc3
testing.(*M).Run(0xc2080468c0, 0xee9680)
	/usr/local/go/src/testing/testing.go:485 +0xe1
main.main()
	github.com/influxdb/influxdb/meta/_test/_testmain.go:270 +0x28d

goroutine 6 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x2d
created by os/signal.initÂ·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x43

goroutine 91 [chan receive]:
testing.(*T).Parallel(0xc2080e0c60)
	/usr/local/go/src/testing/testing.go:413 +0xb6
github.com/influxdb/influxdb/meta_test.TestStore_Open_ErrStoreOpen(0xc2080e0c60)
	/home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/meta/store_test.go:25 +0x3f
testing.tRunner(0xc2080e0c60, 0xedf590)
	/usr/local/go/src/testing/testing.go:447 +0x134
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xd4e

goroutine 93 [chan receive]:
testing.(*T).Parallel(0xc2080e0cf0)
	/usr/local/go/src/testing/testing.go:413 +0xb6
github.com/influxdb/influxdb/meta_test.TestStore_Open_ErrTooManyPeers(0xc2080e0cf0)
	/home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/meta/store_test.go:36 +0x44
testing.tRunner(0xc2080e0cf0, 0xedf5a8)
	/usr/local/go/src/testing/testing.go:447 +0x134
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xd4e

goroutine 92 [chan receive]:
testing.funcÂ·007()
	/usr/local/go/src/testing/testing.go:559 +0x96
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:560 +0xec9
```

**Comment by jwilder** [comment]
The first panic should be fixed by 3b0b227.  Looking into the second one.

**Comment by otoolep** [comment]
@jwilder -- just saw this:
```
=== RUN TestCluster_Restart
[metastore] 2015/08/10 23:28:31 adding new raft peer: nodeId=2 addr=127.0.0.1:53402
[metastore] 2015/08/10 23:28:31 adding new raft peer: nodeId=3 addr=127.0.0.1:48200
panic: test timed out after 10m0s

goroutine 605 [running]:
testing.funcÂ·008()
	/usr/local/go/src/testing/testing.go:681 +0x12f
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x4b

goroutine 1 [chan receive, 9 minutes]:
testing.RunTests(0xb0a108, 0xcc6be0, 0x6d, 0x6d, 0xc20802b701)
	/usr/local/go/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc2080468c0, 0xcd1400)
	/usr/local/go/src/testing/testing.go:485 +0x6c
main.main()
	github.com/influxdb/influxdb/meta/_test/_testmain.go:270 +0x1d5

goroutine 6 [syscall, 10 minutes]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.initÂ·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 91 [chan receive, 9 minutes]:
testing.(*T).Parallel(0xc2080ec6c0)
	/usr/local/go/src/testing/testing.go:413 +0x7f
github.com/influxdb/influxdb/meta_test.TestStore_Open_ErrStoreOpen(0xc2080ec6c0)
	/home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/meta/store_test.go:25 +0x2e
testing.tRunner(0xc2080ec6c0, 0xcc7330)
	/usr/local/go/src/testing/testing.go:447 +0xbf
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xa8b
```

**Comment by otoolep** [comment]
This may be something else, of course.

**Comment by jwilder** [comment]
Fixed via #3650
